app/scripts/data-fetchers/DataFetcher.js

```diff
@@ -64,15 +108,16 @@ export default class DataFetcher {
     });
   }
 
+  /**
+   * Obtain tileset infos for all of the tilesets listed
+   * @param {(info: TilesetInfo) => void} finished - A callback that will be called
+   */
   tilesetInfo(finished) {
     // if this track has a url, server and filetype
     // then we need to register those with the server
-    if (
-      this.dataConfig.server &&
-      this.dataConfig.url &&
-      this.dataConfig.filetype
-    ) {
-      return this.registerFileUrl(this.dataConfig)
+    const { server, url, filetype, coordSystem } = this.dataConfig;
+    if (server && url && filetype) {
+      return this.registerFileUrl({ server, url, filetype, coordSystem })
```

Comment:
TS can't narrow these types when you have access like this, but destructuring is ok.